docs/user/guide-intermediate.md

```diff
@@ -0,0 +1,323 @@
+# Guide to intermediate level Cursorless
+
+This document will try to:
+
+- Highlight the most useful actions and modifiers in Cursorless so you know which ones to focus on learning first.
+- Bring together a few important concepts that underlie a lot of Cursorless behavior.
+- Walk through some example editing operations that are common but perhaps tricky to a beginner.
+
+This document assumes that you understand the very basics of cursorless, like what commands `"take air"` and `"chuck line"` do.
+If you don't have the basics down, please read some of [the main documentation page](README.md) and/or watch [the Cursorless tutorial videos](https://www.youtube.com/watch?v=5mAzHGM2M0k&list=PLXv2sppxeoQZz49evjy4T0QJRIgc_JPqs).
+
+It is planned for this guide to be expanded to both basic and advanced levels.
+
+## Important actions
+
+Here are some of the most useful actions.
+Parts in square braces (`[]`) are optional.
+Slashes (`/`) show alternatives.
+Targets are represented by `T` with a possible digit.
+
+- Selection Manipulation
+  - `"take T"`: set selection.
+  - `"pre/post T"`: set selection before/after T.
+- Clipboard
+  - `"paste to/before/after T"`.
+  - `"carve/copy T"`: cut/copy T.
+- Core Changers
+  - `"bring T"`: insert a copy of T at the cursor/selection.
+  - `"bring T1 to T2"`: replace T2 with T.
+  - `"bring T1 before/after T2"`: insert a copy of T1 before/after T2, including appropriate delimiters.
+  - `"move T1 [to/before/after T2]"`: like `"bring"`, but moves instead of copies.
+  - `"chuck T"`: delete T and appropriate delimiter.
+  - `"change T"`: delete T and set cursor(s) to where T was; delimiters are unchanged.
+  - `"drink/pour T"`: edit new line before/after T.
+- IDE-related
+  - `"follow T"`: open URL or go to definition of T.
+  - `"quick fix T"`: show quick fix for T.
+  - `"reference T"`: show references to T.
+
+## Important modifiers
+
+Here are some very useful modifiers:
+
+- Syntactic scopes ([reference](README.md#syntactic-scopes)) are great for expanding the target to what you want to operate on, like the containing function.
+  - `"arg"` and `"item"`: often used to involve commas, like `"chuck item air"` or `"bring air after arg bat"` that take care of commas.
+  - `"funk"` for function.
```

Comment:
Considering that we are handling multiple languages, there might be no perfect phrasing for what you mean by function declaration, but maybe it is better to phrase it as a function definition or function implementation.

To a Java/C++/C# person, function declaration means something that I think you do not mean.

```c++
void f(); // function declaration; you can have many of these
void f() {} // function definition; you can only have one of these
```

Even typescript has stuff like `declare function getWidget(n: number): Widget;` ([source](https://www.typescriptlang.org/docs/handbook/declaration-files/by-example.html#overloaded-functions)), which is not picked up by "visualize funk".

Reply:
That's fair; just didn't want people to think it could mean the function in a function call. We've had that confusion before, but not sure how common it is. Maybe there's a better term we could use? Maybe "function definition/declaration"?